tests/test_optimizer_dryruns.py

```diff
@@ -148,6 +163,41 @@ def test_clouds_not_enabled(monkeypatch):
                                enabled_clouds=[sky.AWS()])
 
 
+def test_instance_type_mismatches_cpus(monkeypatch):
+    bad_instance_and_cpus = [
+        # Actual: 8
+        ('m6i.2xlarge', 4),
+        # Actual: 2
+        ('c6i.large', 4),
+    ]
+    for instance, cpus in bad_instance_and_cpus:
+        with pytest.raises(ValueError) as e:
+            _test_resources_launch(monkeypatch,
+                                   sky.AWS(),
+                                   instance_type=instance,
+                                   cpus=cpus)
+        assert 'does not have the requested number of vCPUs' in str(e.value)
+
+
+def test_instance_type_matches_cpus(monkeypatch):
+    _test_resources_launch(monkeypatch,
+                           sky.AWS(),
+                           instance_type='c6i.8xlarge',
+                           cpus=32)
+    _test_resources_launch(monkeypatch,
+                           sky.Azure(),
+                           instance_type='Standard_E8s_v5',
+                           cpus=8)
+    _test_resources_launch(monkeypatch,
+                           sky.GCP(),
+                           instance_type='n1-standard-8',
+                           cpus='7+')
+    _test_resources_launch(monkeypatch,
+                           sky.AWS(),
+                           instance_type='g4dn.2xlarge',
+                           cpus=8)
```

Comment:
Maybe add the `8.0` and `'8'` variants for the last resource.

Reply:
Fixed.